subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -78,25 +78,45 @@
         }
     }
 
-    private synchronized Spec<FileTreeElement> getDefaultExcludeSpec(boolean caseSensitive) {
-        Spec<FileTreeElement> specs = defaultExcludeSpecs.get(caseSensitive);
-        List<String> defaultExcludes = Arrays.asList(DirectoryScanner.getDefaultExcludes());
-
-        if (specs == null) {
-            specs = updateDefaultExcludeCache(defaultExcludes, caseSensitive);
-        } else if (!previousDefaultExcludes.equals(defaultExcludes)) {
-            specs = updateDefaultExcludeCache(defaultExcludes, caseSensitive);
+    private synchronized Spec<FileTreeElement> getDefaultExcludeSpec(CaseSensitivity caseSensitivity) {
+        String[] defaultExcludes = DirectoryScanner.getDefaultExcludes();
+        if (defaultExcludeSpecCache.isEmpty()) {
+            updateDefaultExcludeSpecCache(defaultExcludes);
+        }
+        if (!Arrays.equals(previousDefaultExcludes, defaultExcludes)) {
```

Comment:
On the first call to the method, I want to initialize the default excludes cache. That is the purpose of `updateDefaultExcludeSpecCache`.

I can add an `else` here, since the if condition always evaluates to false when the first if condition evaluates to true. Do you think that would be easier to understand?

Reply:
I added the `else`.

Reply:
I think this is definitely better. Without the `else` someone looking at the code would think that sometimes it's possible to call `updateDefaultExcludeSpecCache(defaultExcludes)` twice.